client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -998,14 +1033,22 @@ const ConnectedPlanFeatures2023Grid = connect(
 					isWpcomEnterpriseGridPlan( plan ) && planConstantObj.getPathSlug
 						? planConstantObj.getPathSlug()
 						: planObject?.product_name_short ?? '';
-				const storageOptions =
+				const storageFeatures =
 					( planConstantObj.get2023PricingGridSignupStorageOptions &&
 						planConstantObj.get2023PricingGridSignupStorageOptions(
 							showLegacyStorageFeature,
 							isCurrentPlan
 						) ) ||
 					[];
+				const storageOptions = storageFeatures.map( ( feature ) => {
```

Comment:
The comparison grid has the same `storageOptions` structure from previous [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/components/plan-comparison-grid.tsx#L893). I believe we should be updating that end too to use the same updated structure we are putting together here and ensure it renders the default correctly with the new data structure. 

The two are also merged in the other PR that migrates plan properties out of the components.

Reply:
>  I believe we should be updating that end too to use the same updated structure we are putting together here and ensure it renders the default correctly with the new data structure.

I don't mean to pass `storageOptions` from here to the comparison grid. Just to refactor however it is computing and using storage options to match this new type.

Reply:
Actually, I am sorry, maybe I do mean to reuse this same value there. This might inform this type changing a little. ComparisonGrid needs the actual `FeatureObject` to access `getCompareTitle` (while features grid needs the `getTitle` one). So instead of abstracting things away from `FeatureObject`, let's may consider adding the actual object in the options structure e.g.

1. changing the type from `{ key: string, planDefault: boolean, title: string }` to `{ key: string, planDefault: boolean, feature: FeatureObject }`

2. changing Comparison end [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/components/plan-comparison-grid.tsx#L893) to just access this directly when restructuring the features:

from 

```
			const [ storageOption ] =
				planObject.get2023PricingGridSignupStorageOptions?.( showLegacyStorageFeature ) ?? [];
			planStorageOptionsMap[ planName ] = storageOption;
```

to 

```
			const [ storageOption ] =
				planObject.storageOptions.find( (option) => option.planDefault );
			planStorageOptionsMap[ planName ] = storageOption;
```

I am guessing the rest should just work as normal on Comparison end. Then on this end, just access the title via `feature.getTitle()` when needed.

Reply:
Just give this some thought and let me know if there's a better alternative. I'm mostly thinking about syncing the data structures/concepts being used across these components to avoid ambiguity.

Reply:
> Just give this some thought and let me know if there's a better alternative

I agree with this direction. I was confused by the two different ways of accessing storage titles in the features grid and comparison grid as well. I was reluctant to make the changes because, after some cursory investigation, I couldn't tell whether or not there was some reasoning for doing so.

After hearing that it's something that you're covering in your refactor, I'm happy to sync the changes in this PR. As for the actual shape of our data structure, I don't have a strong opinion. I care more about having a standardized way of accessing storage labels.

Resolved in [8be4eb7](https://github.com/Automattic/wp-calypso/pull/79041/commits/8be4eb704413106d434f0e0da9ebde212d82e4f2)